File: src/main/java/fop/context/impl/ApplicationContextMergeIgnoreStrategy.java
package fop.context.impl;

import java.util.Set;

import fop.context.ApplicationContextKey;

public class ApplicationContextMergeIgnoreStrategy extends AbstractApplicationContextMergeStrategy {

    public ApplicationContextMergeIgnoreStrategy()
    {
        super(null);
    }
    
    public ApplicationContextMergeIgnoreStrategy(Set<String> keys)
    {
        super(keys);
    }
    
    @Override
    public <T> T merge(ApplicationContextKey<T> key, T oldValue, T newValue)
    {
        boolean keySetAvailable = super.keySetAvailable();
        T mergeValue = null;
        
        if(!keySetAvailable)
        {
            mergeValue = oldValue;
        }
        else if(super.keyAvailable(key))
        {
            mergeValue = oldValue;
        }
        else
        {
            mergeValue = newValue;
        }
        
        return mergeValue;
    }

}

File: src/main/kotlin/io/github/funofprograming/context/impl/ApplicationContextMergeIgnoreStrategy.kt
package io.github.funofprograming.context.impl

import io.github.funofprograming.context.Key

class ApplicationContextMergeIgnoreStrategy(keys: Set<Key<*>>?=null) : AbstractApplicationContextMergeStrategy(keys) {

    override fun <T> merge(key: Key<T>, oldValue: T?, newValue: T?): T? {
        return if (!super.keySetAvailable() || super.keyAvailable(key)) oldValue else newValue
    }
}